### PR TITLE
Remove xlrd benchmark after xlrd was updated to 2.0.1 on conda-forge

### LINKED ISSUE #39076: CI: Benchmarks in CI are failing because xlrd was updated to 2.0.1
Raises in Benchmark run (for example https://github.com/pandas-dev/pandas/runs/1674714425) because it seems xlrd was updated on conda-forge
https://anaconda.org/conda-forge/xlrd

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #39076
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Is pinning here the right thing to do in environment.yml?

cc @jreback

### CODE REVIEW

**Comment by phofl** [comment]
Alternatively we could remove this benchmark

**Review by jreback — CHANGES_REQUESTED** [review]
NO, do not pin this. just remove the benchmark

**Comment by phofl** [comment]
Done

**Comment by phofl** [comment]
Have to backport this too.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great, backporting as well to avoid 1.2.x CI failures

**Comment by phofl** [comment]
@jreback greenish based on the changes here. Azure will take a while because of the new master commits

**Comment by jreback** [comment]
thanks @phofl

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

File: asv_bench/benchmarks/io/excel.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def time_write_excel(self, engine):
 
 class ReadExcel:
 
-    params = ["xlrd", "openpyxl", "odf"]
+    params = ["openpyxl", "odf"]
```
Comment (jorisvandenbossche) [inline asv_bench/benchmarks/io/excel.py:43]: We could also rather keep this, but with writing an xls file (in addition to the xlsx file) to read with xlrd?

**Pull request merged**